### PR TITLE
Adds claim api for brave wallet (uplift to 1.16.x)

### PR DESCRIPTION
**Pull request by NejcZdovc**
Uplift of #6952
Resolves brave/brave-browser#12313

Approved, please ensure that before merging: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [ ] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by kylehickinson — APPROVED** [review]

**Comment by kjozwiak** [comment]
Restarting `macOS` via https://ci.brave.com/job/pr-brave-browser-claim-ios-16-macos/3/ as `init` failed as per https://ci.brave.com/job/pr-brave-browser-claim-ios-16-macos/2/execution/node/111/log/.

```
15:18:14  0> Failed to fetch file gs://chromium-clang-format/62bde1baa7196ad9df969fc1f06b66360b1a927b for src/buildtools/mac/clang-format, skipping. [Err: ]
15:18:14  Failed to fetch file gs://chromium-clang-format/62bde1baa7196ad9df969fc1f06b66360b1a927b for src/buildtools/mac/clang-format. [Err: ]
15:18:14  Downloading 1 files took 0.301955 second(s)
15:18:14  Error: Command 'vpython src/third_party/depot_tools/download_from_google_storage.py --no_resume --no_auth --bucket chromium-clang-format -s src/buildtools/mac/clang-format.sha1' returned non-zero exit status 1 in /Users/jenkins/jenkins/workspace/pr-brave-browser-claim-ios-16-macos
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.16.x` approved after deliberating with @brave/uplift-approvers.

**Pull request merged**
